### ISSUE TITLE
Change for loops to map

### ISSUE THREAD

**Issue by superhackerboy** [superhackerboy/totally-random#41]
As @Gridonyx has mentioned, `.map()` iterates through arrays much faster. Let's make these changes to all of the methods where applicable.

**Comment by Gridonyx** [comment]
Working on this now

**Comment by Gridonyx** [comment]
Noticed a bug in `.array` that I'm also fixing, with how it is currently handling `unique = true` it could output an array length less than the specified count

**Comment by Gridonyx** [comment]
This feels like an absolutely disgusting solution but it does fix the bug:
```js
array(arr, count = 1, unique = false) {
  if (arr.length < count && unique) throw new Error(
    "Array length must be larger than count if unique = true"
  );

  let newArr = [...Array(count)].map(
    () => arr[this.to(arr.length - 1)]
  );
  
  if (unique) {
    newArr = [...new Set(newArr)];

    if (newArr.length < count) {
      while (newArr.length < count) {
        const addElem = arr[this.to(arr.length - 1)];

        if (newArr.includes(addElem))
          continue;
        else newArr.push(addElem);
      }
    }
  }

  return count === 1 ? newArr[0] : newArr;
}
```

**Comment by superhackerboy** [comment]
Yeah just make sure you leave comments if it's hard to follow. Also don't forget to add `countError`!

**Comment by Gridonyx** [comment]
I'm going to set trying to resolve the bug aside for now and work on getting things updated to use `.map()`
